### ISSUE TITLE
Wishlist: "fake init".

### ISSUE THREAD

**Issue by vivainio** [fsharp/FAKE#1465]
### Description

Would be nice to get a reasonable starting point by a command line invocation, instead of fetching examples around the web. The flow would be:

"paket convert-from-nuget"
"paket add nuget FAKE"
.\packages\FAKE\tools\FAKE.exe init

This would create the build.cmd and Build.fsx, filling the Build.fsx with a minimal stereotypical recipes (maybe just running msbuild on top level .sln)

Customers using this would mostly be small console apps created with the raw Visual Studio "F# Console App" wizard, and wanting to upgrade from that to get more sophisticated deploy rules etc.

**Comment by rmunn** [comment]
Good idea, but don't forget about a `build.sh` file as well so that the project can be cross-platform and build on Linux and OS X. The one from https://github.com/fsprojects/ProjectScaffold would be a good one to copy.

In the meantime, until this idea is implemented, a good workaround is to just clone https://github.com/fsprojects/ProjectScaffold and tweak their build script to your needs; it's quite complete.

**Comment by vivainio** [comment]
FYI, I have filed a related wishlist item to Forge https://github.com/fsharp-editing/Forge/issues/43

**Comment by matthid** [comment]
Thinking about this ;)

**Comment by vanceism7** [comment]
I also thought about `fake init` when I was getting started with fake. I feel like at least a minimal setup of a build.fsx script would be good. Have some of the core #r references at the beginning + #load intellisense, maybe setup an initial build and clean target (they can be empty), and an option to set a custom name of the script. 

E.g
'fake init mybuild' would name the script mybuild.fsx, load intellisense from the correct directory, etc. I think that alone would make fake a lot more user-friendly to start up with

**Comment by DmitrySenin** [comment]
Let me also offer something which I think will really help with the initial setup.
I'm quite new to Fake - using it 2-3 days - and it was not really easy to create even a basic setup to start implementing my first build script. Mainly it's connected with the fact that I had to spend extra time understanding what Paket and the `dotnet` tool are.
To my mind, having a setup script would be a great alternative to all the ways of [installing Fake](https://fake.build/fake-gettingstarted.html#Install-FAKE). From this perspective, I really like Cake's approach(it's not an advertisement 😆 ) whereas they offer a PowerShell script to hide everything happening undercover(they also provide a Shell script but I'm not a Linux user).
So I prepared such a script and pushed it [here](https://github.com/DmitrySenin/FakeBootstraper). If anybody found my proposal interesting then, please, take a look and share your thoughts.
Thanks

**Comment by matthid** [comment]
@DmitrySenin Thanks for your feedback! 
Actually you can already install it in two lines:
- Install chocolatey: `Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))`
- Install fake `choco install fake -pre`

Obviously we can add this as single line to the docs if that would help.

Also just a word of warning that your script is already obsolete because it is using the "old" (fake 4) runner and not the "new" fake 5 netcore one.

**Comment by DmitrySenin** [comment]
@matthid ,
The main reason why I didn't use `chocolotey` is I don't want to have any global dependencies to use Fake. My main argument against it is that, to my mind, everything should be kind of "specific" to a repo. That, I think, enables everyone to drop all a project's dependencies by just removing its repo from one's PC.
As for fake 4, as I understand there is not stable version of fake 5 yet that's why I'm using fake 4. But many thanks for your warning.

**Comment by matthid** [comment]
@DmitrySenin Yes understood, actually I hope stuff will be a lot simpler very soon because of global dotnet tools. Which means you can do something like `dotnet tool install --global fake` (at least something along those lines) and then just use `fake <..>`.

If you for some reason strictly want one fake per project I'd suggest to use paket, a dummy project file and manage fake via `clitool` in your dependencies (see https://github.com/matthid/fake-bootstrap/tree/paket_clitool for an example)

> As for fake 4, as I understand there is not stable version of fake 5 yet that's why I'm using fake 4.

Basically I feel like it's just a matter of coordination. I'd consider it stable at this point (not doing any breaking changes - only if they are not avoidable for some reason). The only thing is that is has not been tested by a wide audience - but that will be true for the "stable" release on release date as well ;)

But yes at the moment we have kind of a nasty bug in rc011 (intellisense not working), but rc008 should be working and be a good release (besides being a bit slow)

I hope to have a working release out tomorrow.
